rclpy/src/rclpy/names.cpp

```diff
@@ -0,0 +1,289 @@
+  auto topic_deleter = [&](char * topic) {allocator.deallocate(topic, allocator.state);};
+  auto expanded_topic = std::unique_ptr<char, decltype(topic_deleter)>(
+    output_cstr, topic_deleter);
+
+  rcutils_ret = rcutils_string_map_fini(&substitutions_map);
+  if (rcutils_ret != RCUTILS_RET_OK) {
+    throw RCUtilsError("failed to fini string map");
+  }
+
+  if (ret == RCL_RET_BAD_ALLOC) {
+    rcl_reset_error();
+    throw std::bad_alloc();
+  } else if (  // NOLINT
+    ret == RCL_RET_TOPIC_NAME_INVALID ||
+    ret == RCL_RET_UNKNOWN_SUBSTITUTION)
+  {
+    std::string error_text{"topic name '"};
+    error_text += topic;
+    error_text += "' is invalid";
+    throw py::value_error(append_rcl_error(error_text));
+  } else if (ret == RCL_RET_NODE_INVALID_NAME) {
+    std::string error_text{"node name '"};
+    error_text += node_name;
+    error_text += "' is invalid";
+    throw py::value_error(append_rcl_error(error_text));
+  } else if (ret == RCL_RET_NODE_INVALID_NAMESPACE) {
+    std::string error_text{"node namespace '"};
+    error_text += node_namespace;
+    error_text += "' is invalid";
+    throw py::value_error(append_rcl_error(error_text));
+  } else if (ret != RCL_RET_OK) {
+    throw RCLError("failed to do name expansion");
+  }
+
+  return std::string{expanded_topic.get()};
+}
+
+std::string
+remap_topic_name(py::capsule pynode, const char * topic_name)
+{
+  auto node = static_cast<rcl_node_t *>(
+    rclpy_handle_get_pointer_from_capsule(pynode.ptr(), "rcl_node_t"));
+  if (!node) {
+    throw py::error_already_set();
+  }
+
+  // Get the node options
+  const rcl_node_options_t * node_options = rcl_node_get_options(node);
+  if (nullptr == node_options) {
+    throw RCLError("failed to get node options");
+  }
+
+  const rcl_arguments_t * global_args = nullptr;
+  if (node_options->use_global_arguments) {
+    global_args = &(node->context->global_arguments);
+  }
+
+  char * output_cstr = nullptr;
+  rcl_ret_t ret = rcl_remap_topic_name(
+    &(node_options->arguments),
+    global_args,
+    topic_name,
+    rcl_node_get_name(node),
+    rcl_node_get_namespace(node),
+    node_options->allocator,
+    &output_cstr);
+  if (ret != RCL_RET_OK) {
+    throw RCLError("failed to remap topic name");
+  }
+
+  if (nullptr == output_cstr) {
+    // no remapping found
+    return std::string{topic_name};
+  }
+
+  auto output = std::string{output_cstr};
+  node_options->allocator.deallocate(output_cstr, node_options->allocator.state);
+  return output;
+}
+
+std::string
+resolve_name(py::capsule pynode, const char * topic_name, bool only_expand, bool is_service)
+{
+  auto node = static_cast<rcl_node_t *>(
+    rclpy_handle_get_pointer_from_capsule(pynode.ptr(), "rcl_node_t"));
+  if (!node) {
+    throw py::error_already_set();
+  }
+
+  const rcl_node_options_t * node_options = rcl_node_get_options(node);
+  if (nullptr == node_options) {
+    throw RCLError("failed to get node options");
+  }
+
+  char * output_cstr = NULL;
```

Comment:
@sloretz nit: 
```suggestion
  char * output_cstr = nullptr;
```

Reply:
NULL -> nullptr in dcfc7e28d67657fcb329704b7e78e3a88ef38d6c